src/protocols/dhcpv4/encode.c

```diff
@@ -106,39 +108,40 @@ static ssize_t encode_value(uint8_t *out, size_t outlen,
 	fr_proto_da_stack_build(da_stack, vp ? vp->da : NULL);
 
 	FR_PROTO_STACK_PRINT(da_stack, depth);
-	FR_PROTO_HEX_DUMP(out, (p - out), "Value");
+	FR_PROTO_HEX_DUMP(data_ptr, len, "Value");
 
-	return p - out;
+	return len;
```

Comment:
Manually recording the len value makes this much more complex than it needs to be.  The pattern should be take a copy of the dbuff when entering the function, and set `start = p`.  Then advance the parent at the end.  Feel free to add a debuff function to do that, or you can just advance it by fr_dbuff_used

Reply:
I don't think there'll be a large performance penalty for this.  In the majority of cases the fr_dbuff_t structures will all be on the stack, and they'll likely be close enough together that at least some of the time they'll be in the same 64 byte cache line so copying between them  should only be slightly slower than a register to register copy.

If you think about it, it's very similar to what we do anyway (setting p to out, and q to p + len), it's just the structure involved is larger.

Reply:
Agreed. Switched to non-advancing dbuff.